types/web3/web3-tests.ts

```diff
@@ -10,6 +11,23 @@ const contractAddress = "0xde0B295669a9FD93d5F28D9Ec85E40f4cb697BAe";
 // --------------------------------------------------------------------------
 const web3 = new Web3();
 const myProvider = new web3.providers.HttpProvider("http://localhost:5454");
+
+const responses: JsonRPCResponse[] = [];
+const errors: Error[] = [];
+class FakeHttpProvider implements Provider {
+    send(
+        payload: JsonRPCRequest,
+        callback: (e: Error, val: JsonRPCResponse) => void
+    ) {
+        const response = responses.shift();
```

Comment:
You must have some code to create a response, right? Actually the line should be.
```typescript
const response: JsonRPCResponse = createFakeResponse()
```
You are assigning `undefined` to `response` which is illegal in any regard. Also what is this simulating? An error or a success? In case of success, `error` must be `null`, not `undefined`.

Reply:
The response are created manually. I'm essentially converting https://github.com/ethereum/web3.js/blob/1.0/test/helpers/FakeHttpProvider.js to TypeScript. This is what I am using to create a response: 

```javascript
injectResult(result: any) {
        let response = {
            jsonrpc: '2.0',
            id: this.countId,
            result: result
        }

        this.response.push(response);
}

 provider.injectResult('0x5550000000000000000000000000000000000000000000000000000000000001')
```